tests/fgpyo/util/test_types.py

```diff
@@ -1,11 +1,28 @@
+import sys
 from typing import Iterable
 from typing import List
+from typing import Optional
 from typing import Sequence
+from typing import Union
 
 from fgpyo.util import types
+from fgpyo.util.inspect import NoneType
 
 
 def test_is_listlike() -> None:
     assert types.is_list_like(List[str])
     assert types.is_list_like(Iterable[str])
     assert types.is_list_like(Sequence[str])
+    assert not types.is_list_like(str)
+
+
+def test_is_optional() -> None:
+    assert types._is_optional(Union[str, NoneType])
+    assert types._is_optional(Optional[str])
+    assert not types._is_optional(str)
```

Comment:
**suggestion** Test a couple union types that don't meet our criteria

```suggestion
    assert types._is_optional(Optional[str])
    
    assert not types._is_optional(Union[str, int])
    assert not types._is_optional(Union[str, int, None])
    assert not types._is_optional(str)
```

Reply:
NB: for brevity (and to catch multiple failures in a single `pytest` invocation), you could use `pytest.mark.parameterize` over the tested types